src/abc_atlas_access/abc_atlas_cache/cloud_cache.py

```diff
@@ -1113,15 +1211,19 @@ def _download_file(self, file_attributes: CacheFileAttributes) -> bool:
             was_downloaded = True
             response = self.s3_client.get_object(Bucket=self._bucket_name,
                                                  Key=str(obj_key))
+            hasher = hashlib.md5()
```

Comment:
Is this significantly faster than downloading the file then calling `file_hash_from_path()` on the downloaded file?

I'm a little wary of implementing the hashing independently in two places. I cannot promise that we won't be maniacs and decide in ~ 3 years that we actually did not want the MD5 hash. It would be good to have the hashing algorithm specified in only one place.

Unless you have found that this is much faster.

Reply:
It's sort of a free parallelism (at best). At mid, it makes the progress bar a bit more user informative. As in, my file is ready and a link to it in my lap once the progress bar finishes. I can move it back as I do take your point about having the code in separate places. What we could do is create a light hasher class that has the check file method and allows for just the update chunk. That said, we are basically just wrapping the hasher class anyway (that is in the file path). I'll think about it and we can talk it over what is worth doing.

Reply:
I'll revert to just having the file be checked.